### PR TITLE
MARKET-1332 Popover

### PR DESCRIPTION
**Pull request by StasivPavlo**
image
image

### CODE REVIEW

File: components/bl-popover-component/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+| text        | String                                  |               | Text Logic         | YES          | YES        | Allows to determine the text of popover                                |
+| buttonLabel | String                                  |               | Button Label Logic | YES          | YES        | Allows to determine the label of button                                |
+| position    | Select ["Top", "Right","Bottom","Left"] | "Top"         | Position Logic     | NO           | YES        | Allows to determine the position of the popover relative to the button |
+| disabled    | Boolean                                 | `false`       | Disabled Logic     | NO           | YES        | Allows determining whether a button is disabled or not                 |
+
+## Events
+
+| Name     | Triggers                   | Context Blocks     |
+|----------|----------------------------|--------------------|
+| On Click | when user clicks on button | `Is Open: boolean` |
```
Comment (addKiwi) [inline components/bl-popover-component/README.md:21]:
```suggestion
| On Click | when user clicks on button | Is Open: `Boolean` |
```

File: components/bl-popover-component/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## Actions
+
+| Action      | Inputs             | Returns |
+|-------------|--------------------|---------|
+| Set Is Open | `boolean: boolean` |         |
```
Comment (addKiwi) [inline components/bl-popover-component/README.md:27]:
```suggestion
| Set Is Open | Boolean: `Boolean` |         |
```

**Review by addKiwi — CHANGES_REQUESTED** [review]

File: components/bl-popover-component/src/helpers.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  const buttonOffset = offset(buttonElement);
+
+  if (position === 'right' || position === 'left') {
+    const buttonHorizontalCenter = buttonElement.clientHeight / 2;
+    const rootHorizontalCenter = root.clientHeight / 2;
+
+    rootTranslateTop = buttonOffset.top - rootOffset.top + buttonHorizontalCenter - rootHorizontalCenter;
+
+    if (position === 'right') {
+      rootTranslateLeft = buttonOffset.left + buttonElement.getBoundingClientRect().width + root.firstChild.clientHeight + rootOffset.left;
```
Comment (v-excelsior) [inline components/bl-popover-component/src/helpers.js:14]: You didn't have eslint warnings?

File: components/bl-popover-component/src/helpers.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+export const translatePopover = (buttonElement, root, position) => {
```
Comment (v-excelsior) [inline components/bl-popover-component/src/helpers.js:1]: Do you write all that logic by yourself? If no, could you share an reference?

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-popover-component/src/helpers.js (hunk 4 above)
Comment (StasivPavlo) [reply components/bl-popover-component/src/helpers.js]: I wrote it myself

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-popover-component/src/Tooltip.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    return () => {
+      document.body.removeChild(root);
+    };
+  }, []);
+
+  const translateHandler = useCallback(() => {
+    root.style.transform = 'translate3d(0px, 0px, 0px)';
+
+    if (contentContainerElement) {
+      const { rootTranslateLeft, rootTranslateTop } = translatePopover(contentContainerElement, root, position);
```
[the remark this answers is not in the document]
Comment (StasivPavlo) [reply components/bl-popover-component/src/Tooltip.js:24]: shiftLeft and shiftTop we get from the property?

File: components/bl-popover-component/src/index.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import { Tooltip } from './Tooltip';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function Popover({ component, eventHandlers, pods }) {
+  const { display, style, classList, position } = component;
+  const { onClick } = eventHandlers;
+
+  const popoverTrigger = pods['popoverTrigger'];
+  const popoverContent = pods['popoverContent'];
```
Comment (v-excelsior) [inline components/bl-popover-component/src/index.js:11]:
```suggestion
const { popoverTrigger, popoverContent }  = pods
```

File: components/bl-popover-component/src/index.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+  const [isOpen, setIsOpen] = useState(false);
+  const contentElement = useRef();
+
+  component.setIsOpen = (boolean) => {
+    setIsOpen(boolean);
+  };
```
Comment (v-excelsior) [inline components/bl-popover-component/src/index.js:18]:
```suggestion
  component.setIsOpen = (isOpen) => setIsOpen(isOpen);
```

**Review by v-excelsior — CHANGES_REQUESTED** [review]

**Pull request merged**